src/wallet/wallet.cpp

```diff
@@ -3038,12 +3038,17 @@ bool CWallet::TopUpKeyPool(unsigned int kpSize)
 
 bool CWallet::GetNewDestination(const OutputType type, const std::string label, CTxDestination& dest, std::string& error)
 {
+    LOCK(cs_wallet);
     error.clear();
     bool result = false;
     auto spk_man = m_spk_man.get();
     if (spk_man) {
-        result = spk_man->GetNewDestination(type, label, dest, error);
+        result = spk_man->GetNewDestination(type, dest, error);
+    }
+    if (result) {
```

Comment:
Does it really matter?

Reply:
No, just that it looks like a leftover where before result could be changed in between. 

Actually I think this would look better without the result variable.

You can resolve this.